src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1518,6 +1518,79 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
             break;
         case NI_Sve_ConditionalSelect:
             return LowerHWIntrinsicCndSel(node);
+        case NI_Sve_SetFfr:
+        {
+            StoreFFRValue(node);
+            break;
+        }
+        case NI_Sve_GetFfrByte:
+        case NI_Sve_GetFfrInt16:
+        case NI_Sve_GetFfrInt32:
+        case NI_Sve_GetFfrInt64:
+        case NI_Sve_GetFfrSByte:
+        case NI_Sve_GetFfrUInt16:
+        case NI_Sve_GetFfrUInt32:
+        case NI_Sve_GetFfrUInt64:
+        {
+            LIR::Use use;
+            bool     foundUse = BlockRange().TryGetUse(node, &use);
+            if (foundUse)
+            {
+                // Remove GetFfr and replace the GetFfr's use with the lvaFfrRegister lclVar.
+                CreateFFRDefIfNeeded(node);
```

Comment:
I wouldn't expect this to be necessary here

Reply:
can you elaborate?

Reply:
We insert the stores into `lvaFfrRegister` after all operations that may modify it -- `setffr` and `LoadFF` -- thus the local itself always has the up-to-date value. So I don't think we should read from it as part of `GetFFR` -- it can just use the already up-to-date FFR value directly.

Reply:
fixed